platform/api-server/api/controllers/gateway-api-controller.js

```diff
@@ -0,0 +1,82 @@
+const discoveryModel = require('../models/discovery-model');
+const path = require("path");
+let ipc = require('node-ipc');
```

Comment:
Ideally, all of the IPC stuff should be in api-server/server.js rather than in individual controllers. The ipc object should then be made available to all controllers and not just to the gateway-api-controller. I will take care of doing that change. Add a TODO here stating that this needs to be moved to server.js